prog/dftb+/prg_dftb/mainio.F90

```diff
@@ -2475,19 +2475,50 @@ subroutine writeDetailedOut1(fd, fileName, tAppendDetailedOut, iDistribFn, nGeoS
       else
         write(fd, "(2A)") 'COMPONENT = ', trim(quaternionName(iSpin))
       end if
-      write(fd, "(/, A)") 'Eigenvalues /H'
-      do iEgy = 1, size(eigen, dim=1)
-        write(fd, formatEigen) (eigen(iEgy, iK, iSpin), iK = 1, nKPoint)
-      end do
-      write(fd, "(/, A)") 'Eigenvalues /eV'
-      do iEgy = 1, size(eigen, dim=1)
-        write(fd, formatEigen) (Hartree__eV * eigen(iEgy, iK, iSpin), iK = 1, nKPoint)
+      write(fd, "(/, A)") 'Eigenvalues /H and fillings'
+      do iK = 1, nKPoint, 4
+        if (nKPoint > 1) then
+          if (nKPoint - iK > 0) then
+            write(fd,"(A,I0,':',I0)")'K-points ',iK,min(iK+3,nKPoint)
+          else
+            write(fd,"(A,I0,':',I0)")'K-point ',iK
+          end if
+        end if
+        do iEgy = 1, size(eigen, dim=1)
+          write(fd, "(I6)", advance = 'no') iEgy
+          do kk = 0, 3
+            if (iK + kk > nKPoint) then
+              exit
+            end if
+            write(fd, formatEigen, advance = 'no') eigen(iEgy, iK+kk, iSpin),&
+                & filling(iEgy, iK+kk, iSpin)
+          end do
+          write(fd, formatEigen)
+        end do
+        write(fd, *)
       end do
-      write(fd, "(/, A)") 'Fillings'
-      do iEgy = 1, nLevel
-        write(fd, formatFilling) (filling(iEgy, iK, iSpin), iK = 1, nKPoint)
+      write(fd, "(/, A)") 'Eigenvalues /eV and fillings'
+      do iK = 1, nKPoint, 4
+        if (nKPoint > 1) then
+          if (nKPoint - iK > 0) then
+            write(fd,"(A,I0,':',I0)")'K-points ',iK,min(iK+3,nKPoint)
+          else
+            write(fd,"(A,I0,':',I0)")'K-point ',iK
+          end if
+        end if
+        do iEgy = 1, size(eigen, dim=1)
+          write(fd, "(I6)", advance = 'no') iEgy
+          do kk = 0, 3
+            if (iK + kk > nKPoint) then
+              exit
+            end if
+            write(fd, formatEigen, advance = 'no') Hartree__eV * eigen(iEgy, iK+kk, iSpin),&
+                & filling(iEgy, iK+kk, iSpin)
+          end do
+          write(fd, formatEigen)
+        end do
+        write(fd, *)
       end do
```

Comment:
I know, that the write routines are not granular enough yet in general, but let's package the eigenvalue printing into a separate private routine, which is then called by `writeDetailedOut1`. Latter is already much beyond the optimal 50-100 statement lines per routine.

Reply:
OK, will push something.